### ISSUE TITLE
`literal_string_with_formatting_args` triggered by `cfg_attr` feature

### ISSUE THREAD

**Issue by messense** [rust-lang/rust-clippy#14041]
### Summary

Beta clippy starts to trigger this in [`pyo3/pyo3`](https://github.com/PyO3/pyo3/actions/runs/12844598072/job/35817782368?pr=4865)

### Lint Name

literal_string_with_formatting_args

### Reproducer

I tried this [code](https://github.com/PyO3/pyo3/blob/ad5f6d404fcf0849ca03cdac2c498ea18e74a588/src/lib.rs):

```rust
#![warn(missing_docs)]
#![cfg_attr(
    feature = "nightly",
    feature(auto_traits, negative_impls, try_trait_v2)
)]
```

I saw this happen:

```
error: this looks like a formatting argument but it is not part of a formatting macro
   --> src/lib.rs:3:22
    |
  3 |       feature = "nightly",
    |  ______________________^
  4 | |     feature(auto_traits, negative_impls, try_trait_v2)
    | |_^
    |
```

I expected to see this happen: no warning/error

### Version

```text
rustc 1.85.0-beta.4 (752fecf80 2025-01-17)
binary: rustc
commit-hash: 752fecf8005cd7a7ae8c0b74d0f8fc3cd3cd86fe
commit-date: 2025-01-17
host: x86_64-unknown-linux-gnu
release: 1.85.0-beta.4
LLVM version: 19.1.7
```

**Comment by y21** [comment]
Probably a duplicate of #13885. That is, the issue is likely not related to cfg_attr, but that it emits a warning on a dummy span.

**Comment by messense** [comment]
@y21 Thanks for the reply, marked as duplicate.
